### DISCUSSION TITLE
How to compile Antlr4 c++ runtime without tests

### DISCUSSION THREAD

**Discussion by ishaangupte21** [antlr/antlr4#3731] [Q&A]
I am trying to cross compile the C++ runtime for ARM on an Intel machine and the tests keep failing (for obvious reasons). Is there any way to compile the runtime without the test rig?

**Comment by mike-lischke** [comment]
The test rig is a Java framework, which runs independently of the C++ build. Not sure how you manage to build the tests along with the C++ build.

**Comment by KvanTTT** [comment]
If I understand you correctly, it should be quite easy. You just have to execute commands from the method [`initRuntime()`](https://github.com/antlr/antlr4/blob/dev/runtime-testsuite/test/org/antlr/v4/test/runtime/cpp/CppRunner.java#L123-L143):

### For Linux

```bash
cd runtime/Cpp
cmake . -DCMAKE_BUILD_TYPE=Release
make
```

### For Windows

```bash
cd runtime/Cpp/runtime
msbuild antlr4cpp-vs2022.vcxproj /p:configuration=Release DLL /p:platform=x64
```
